.gitignore

```diff
@@ -131,3 +131,6 @@ SQLITE_MAX_VARIABLE_NUMBER.cache
 
 # don't ignore the TTF font for H5P
 !h5p-core-18.ttf
+
+# static c extensions cache folder
+cext_cache
```

Comment:
Can you describe why this was necessitated? I would think this dir is only every used in docker

Reply:
ah yes it's because when running in Travis and locally, the path `/cext_cache` that we use in docker is not writable, so I have to change it to the path `cext_cache` in that case. It's from this function: https://github.com/learningequality/kolibri/blob/4a564d952636b8137d70da6c6ca6d33a70072c5c/build_tools/install_cexts.py#L231

Reply:
I'm... not sure I understand. But I don't work with Travis very much. If it's building in Travis "locally", why redefine the cache directory? Would we run into a similar problem if someone were to try and build this locally, without Docker?

I _think_ I found there error here:
https://travis-ci.org/github/learningequality/kolibri/jobs/708571303

Reply:
ah that one was the different error. It was because this script requires a pip version > 19.3.1. The code resolving that issue is here:
https://github.com/learningequality/kolibri/blob/4a564d952636b8137d70da6c6ca6d33a70072c5c/tox.ini#L86-L88
and
https://github.com/learningequality/kolibri/blob/4a564d952636b8137d70da6c6ca6d33a70072c5c/build_tools/install_cexts.py#L257-L264

The error related to the path `/cext_cache` is here: https://travis-ci.org/github/learningequality/kolibri/jobs/708533742#L975
And yes, you are running the script locally or in Travis, and not running as the root user, you will get a permission error when trying to create the path `/cext_cache`. So when that happens, we redefine the path to be `kolibri/cext_cache`